src/OrleansRuntime/GrainDirectory/MultiClusterRegistration/RegistrarManager.cs

```diff
@@ -4,25 +4,24 @@
 
 namespace Orleans.Runtime.GrainDirectory
 {
+    using Orleans.Runtime.Configuration;
+
     /// <summary>
     /// Maps multi-cluster registration strategies to the corresponding registrar
     /// </summary>
     internal class RegistrarManager
     {
         private readonly Dictionary<Type, IGrainRegistrar> registrars = new Dictionary<Type, IGrainRegistrar>();
-
-        public static RegistrarManager Instance { get; private set; }
-
-
-        private RegistrarManager()
-        {
-        }
-
-        public static void InitializeGrainDirectoryManager(LocalGrainDirectory router, int numRetriesForGSI)
+        
+        public RegistrarManager(GrainDirectoryPartition directoryPartition, GlobalSingleInstanceActivationMaintainer gsiActivationMaintainer, GlobalConfiguration globalConfig, Logger logger)
         {
-            Instance = new RegistrarManager();
-            Instance.Register<ClusterLocalRegistration>(new ClusterLocalRegistrar(router.DirectoryPartition));
-            Instance.Register<GlobalSingleInstanceRegistration>(new GlobalSingleInstanceRegistrar(router.DirectoryPartition, router.Logger, router.GsiActivationMaintainer, numRetriesForGSI));
+            this.Register<ClusterLocalRegistration>(new ClusterLocalRegistrar(directoryPartition));
+            this.Register<GlobalSingleInstanceRegistration>(
+                new GlobalSingleInstanceRegistrar(
```

Comment:
As in `SiloStatisticsManager`: It feels like these could be give as parameters in the future. As in not creating instances in constructor using new and configuration options, but giving the instances via constructor and moving the configurations closer to the source. Hene "unmasking" the process. If this makes sense, maybe we could add a comment (and issue?) about it. If not, ignore this.

There are probably other places this line of reasoning could be added.